asf_heat_pump_suitability/analysis/flats_on_fossils/analysis_answer_research_questions.py

```diff
@@ -0,0 +1,462 @@
+        "community_heating",
+        "fossil_fuel_heating",
+    ]
+).null_count() / len(flats_epc_df)
+
+# %%
+flats_epc_df.filter(pl.col("fuel_type").is_null())["community_heating"].value_counts()
+
+# %%
+flats_epc_df.filter(pl.col("fuel_type").is_null())[
+    "MAINHEAT_DESCRIPTION"
+].value_counts()
+
+# %% [markdown]
+# ## Preprocessing
+# Here is an additional preprocessing step to rename the building rise categories for plotting
+
+# %%
+rise_order = [
+    "Low-rise\n(1-3 storeys)",
+    "Medium-rise\n(4-10 storeys)",
+    "High-rise\n(11+ storeys)",
+    "Unknown",
+]
+
+building_rise_dict = {
+    "low-rise": "Low-rise\n(1-3 storeys)",
+    "medium-rise": "Medium-rise\n(4-10 storeys)",
+    "high-rise": "High-rise\n(11+ storeys)",
+}
+
+flats_epc_df = raw_flats_epc_df.with_columns(
+    pl.col("building_rise")
+    .replace_strict(building_rise_dict, default="Unknown")
+    .alias("building_rise_title")
+).with_columns(
+    pl.col("building_rise_title").cast(pl.Enum(rise_order)),
+)
+
+# %% [markdown]
+# ## Answer research questions
+
+# %% [markdown]
+# ## 1. How many flats are there in England, Scotland, and Wales?
+#
+# Information from combined Scotland and England and Wales census data: 6,238,634.
+
+# %%
+census_property_df = get_target.transform_df_target_property_type()
+census_property_df["Flat, maisonette or apartment"].sum()
+
+# %%
+len(flats_epc_df)
+
+# %% [markdown]
+# ## 2. What is the proportion of each flat type (e.g. high rise, low rise)?
+
+# %%
+total_buildings_per_rise_df = (
+    flats_epc_df["building_rise_title"]
+    .value_counts(sort=True, normalize=False)
+    .rename({"count": "total_count"})
+)
+total_buildings_per_rise_df
+
+# %%
+plot_df = (
+    flats_epc_df["building_rise_title"]
+    .value_counts(normalize=True)
+    .sort(by="building_rise_title")
+)
+plot_df
+
+# %%
+plot_df = (
+    flats_epc_df.filter(pl.col("building_rise").is_not_null())["building_rise_title"]
+    .value_counts(normalize=True)
+    .sort(by="building_rise_title")
+)
+plot_df
+
+# %%
+labels = list(
+    zip(plot_df["building_rise_title"].to_list(), (plot_df["proportion"] * 100).round())
+)
+labels = [k + f", {int(v)}%" for k, v in labels]
+
+plt.pie(
+    plot_df["proportion"],
+    labels=labels,
+    startangle=90,
+    counterclock=False,
+    colors=["#0000FF", "#FDB633", "#18A48C"],
+    wedgeprops={"edgecolor": "black", "linewidth": 1},
+)
+plt.title("Proportion of EPC flats by building rise type in GB")
+plt.tight_layout()
+plt.show()
+
+# %%
+# Including null values
+plot_df = (
+    flats_epc_df["building_rise_title"]
+    .value_counts(normalize=True)
+    .sort(by="building_rise_title")
+)
+plt.pie(
+    plot_df["proportion"],
+    labels=plot_df["building_rise_title"],
+    startangle=90,
+    counterclock=False,
+    colors=["#0000FF", "#FDB633", "#9A1BBE", "#18A48C"],
+    wedgeprops={"edgecolor": "black", "linewidth": 1},
+)
+plt.title("Proportion of EPC flats by building rise type in GB")
+plt.show()
+
```

Comment:
```suggestion
plot_df_nations = (
    flats_epc_df
    .filter(pl.col("building_rise").is_not_null())
    .group_by(["COUNTRY", "building_rise_title"])
    .agg(
        pl.count().alias("count")
    )
    .with_columns(
        (pl.col("count") / pl.col("count").sum().over("COUNTRY"))
        .round(3)
        .alias("proportion")
    )
    .sort(["COUNTRY", "building_rise_title"])
)

print(plot_df_nations)
```

doing this for each of the nations might be an easy win for this case.